{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Great, now that we discussed a little let's continue"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Given that the current approach utilized by the authors lacks reproducibility, we will explore an alternative method by leveraging nf-core pipelines for data analysis.\n",
    "\n",
    "##### Please explain, how we will achieve reproducibility for the course  with this approach.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will ensure reproducibility by utilizing nf-core pipelines.<br>\n",
    "These pipelines are characterized by standardized workflows, strict version control, thorough documentation, and strong community support, all of which contribute to reproducibility.<br>\n",
    "By simply providing the input and output in the standardized format outlined in the documentation, the pipeline will run consistently across any infrastructure.<br><br>\n",
    "\n",
    "**Source:**<br>\n",
    "Ewels, P., Magnusson, M., Lundin, S., & Kaller, M. (2020). nf-core: a collection of community-driven bioinformatics pipelines. *Nature Methods*, 17(1), 24-25. https://doi.org/10.1038/s41592-019-0660-0\n",
    "<br><br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### You have successfully downloaded 2 of the fastq files we will use in our study.\n",
    "##### What is the next step if we want to first have a count table and check the quality of our fastq files? What is the pipeline called to do so?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The next step is to run a quality control pipeline, followed by a quantification pipeline.<br>\n",
    "To proceed, we need to install the appropriate pipeline and prepare the input data, including the samplesheet, as outlined in the documentation.<br>\n",
    "The nf-core/rnaseq pipeline can be used for this purpose.<br><br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Analyze the 2 files using an nf-core pipeline."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### What does this pipeline do?\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This tool performs the entire RNA-seq analysis workflow, from quality control to the generation of a count table, ensuring that the analysis is both reproducible and efficient.<br>\n",
    "The individual steps are outlined below.<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Which are the <font color=\"#008080\">main tools</font> that will be used in the pipeline?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. **Quality checks** on raw sequencing reads using <font color=\"#008080\"> FastQC</font> or <font color=\"#008080\">MultiQC</font>.<br>\n",
    "2. **Read Alignment** to a reference genome using an aligner like <font color=\"#008080\">STAR</font> or <font color=\"#008080\"> FastQC</font>HiSAT2</font><br>\n",
    "3. **Quantification** of reads involves measuring gene or transcript expression levels using <font color=\"#008080\"> FastQC</font>featureCounts (which counts reads mapped to genes) or alignment-free methods like <font color=\"#008080\"> Salmon</font> or <font color=\"#008080\">Kallisto</font><br>\n",
    "4. **Post-Alignment Quality Control** using <font color=\"#008080\">RSeQC</font> <br>\n",
    "5. Production of **count matrix** for downstream analysis (differential gene expression analysis) using <font color=\"#008080\">DESeq2</font> or <font color=\"#008080\">edgeR </font><br>\n",
    "6.  <font color=\"#008080\">MultiQC</font>Report **summarizing all quality control and alignment metrics across samples**<br><br>\n",
    "\n",
    "**Source:**<br>\n",
    "Ewels, P., Magnusson, M., Lundin, S., & Kaller, M. (2020). nf-core: a collection of community-driven bioinformatics pipelines. *Nature Methods*, 17(1), 24-25. https://doi.org/10.5281/zenodo.1400710\n",
    "<br><br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### As all other nf-core pipelines, the chosen pipeline takes in a samplesheet as input.\n",
    "\n",
    "##### Use Python and pandas to create the samplesheet for your 2 samples. Feel free to make use of the table you created earlier today.\n",
    "\n",
    "##### Choose your sample names wisely, they must be the connection of the results to the metadata. If you can't find the sample in the metadata later, the analysis was useless."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Patient          Run RNA-seq  DNA-seq condition: Sal Condition: Oxy  \\\n",
      "3       ?  SRR23195511       x      NaN            NaN              x   \n",
      "5       ?  SRR23195516       x      NaN            NaN              x   \n",
      "\n",
      "  Genotype: SNI Genotype: Sham  \n",
      "3           NaN              x  \n",
      "5             x            NaN  \n",
      "\n",
      "\n",
      "Samplesheet:\n",
      "     sample                             fastq_1  \\\n",
      "0  Sham_oxy  SRX19144488_SRR23195511_1.fastq.gz   \n",
      "1   SNI_oxy  SRX19144486_SRR23195516_1.fastq.gz   \n",
      "\n",
      "                              fastq_2 strandedness  \n",
      "0  SRX19144488_SRR23195511_2.fastq.gz         auto  \n",
      "1  SRX19144486_SRR23195516_2.fastq.gz         auto  \n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# read in the data from the previous sheet where we selected the 2 smallest runs from our dataset\n",
    "samplesheet_df = pd.read_csv('filtered_runs_with_header.csv', header=0)\n",
    "\n",
    "df = pd.read_excel('conditions_runs_oxy_project_sorted.xlsx',header=0)\n",
    "\n",
    "# Merge or filter the DataFrames based on the sample identifiers\n",
    "merged_d = df[df['Run'].isin(samplesheet_df['sample'])]\n",
    "\n",
    "# Display the resulting DataFrame\n",
    "print(merged_d)\n",
    "\n",
    "# Create samplesheet using the information of the table \n",
    "data = {\n",
    "    'sample': ['Sham_oxy','SNI_oxy'],\n",
    "    'fastq_1': ['SRX19144488_SRR23195511_1.fastq.gz','SRX19144486_SRR23195516_1.fastq.gz'],\n",
    "    'fastq_2': [ 'SRX19144488_SRR23195511_2.fastq.gz','SRX19144486_SRR23195516_2.fastq.gz'],\n",
    "    'strandedness':['auto','auto']\n",
    "}\n",
    "\n",
    "# Create the samplesheet DataFrame\n",
    "samplesheet_df = pd.DataFrame(data)\n",
    "\n",
    "# Display the DataFrame\n",
    "print('')\n",
    "print('')\n",
    "print('Samplesheet:')\n",
    "print(samplesheet_df)\n",
    "\n",
    "# Save the samplesheet DataFrame to a CSV file\n",
    "samplesheet_df.to_csv('samplesheet.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In a samplesheet the samples are usually named by their condition and the genotype.<br>\n",
    "We can look these up in the dataframe that we just computed.<br>\n",
    "Run \"SRR23195511\" has genotype \"Sham\" and condition \"oxy\" and the sample \"SRR23195516\" has genotype \"SNI\" and condition \"oxy\"\n",
    "<br><br>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mNextflow 24.04.4 is available - Please consider updating your version to it\u001b[m\n",
      "N E X T F L O W  ~  version 23.10.0\n",
      "Launching `https://github.com/nf-core/rnaseq` [mighty_wozniak] DSL2 - revision: 33df0c05ef [master]\n",
      "\u001b[33mWARN: Access to undefined parameter `monochromeLogs` -- Initialise it to a default value eg. `params.monochromeLogs = some_value`\u001b[39m\u001b[K\n",
      "\n",
      "\n",
      "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
      "                                        \u001b[0;32m,--.\u001b[0;30m/\u001b[0;32m,-.\u001b[0m\n",
      "\u001b[0;34m        ___     __   __   __   ___     \u001b[0;32m/,-._.--~'\u001b[0m\n",
      "\u001b[0;34m  |\\ | |__  __ /  ` /  \\ |__) |__         \u001b[0;33m}  {\u001b[0m\n",
      "\u001b[0;34m  | \\| |       \\__, \\__/ |  \\ |___     \u001b[0;32m\\`-._,-`-,\u001b[0m\n",
      "                                        \u001b[0;32m`._,._,'\u001b[0m\n",
      "\u001b[0;35m  nf-core/rnaseq v3.16.0-g33df0c0\u001b[0m\n",
      "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
      "\u001b[1mCore Nextflow options\u001b[0m\n",
      "  \u001b[0;34mrevision        : \u001b[0;32mmaster\u001b[0m\n",
      "  \u001b[0;34mrunName         : \u001b[0;32mmighty_wozniak\u001b[0m\n",
      "  \u001b[0;34mcontainerEngine : \u001b[0;32mdocker\u001b[0m\n",
      "  \u001b[0;34mlaunchDir       : \u001b[0;32m/home/maikenaegele/Computational Workflow/Sheet 2/Maike_Naegele_sheet2\u001b[0m\n",
      "  \u001b[0;34mworkDir         : \u001b[0;32m/home/maikenaegele/Computational Workflow/Sheet 2/Maike_Naegele_sheet2/work\u001b[0m\n",
      "  \u001b[0;34mprojectDir      : \u001b[0;32m/home/maikenaegele/.nextflow/assets/nf-core/rnaseq\u001b[0m\n",
      "  \u001b[0;34muserName        : \u001b[0;32mmaikenaegele\u001b[0m\n",
      "  \u001b[0;34mprofile         : \u001b[0;32mdocker\u001b[0m\n",
      "  \u001b[0;34mconfigFiles     : \u001b[0;32m\u001b[0m\n",
      "\n",
      "\u001b[1mInput/output options\u001b[0m\n",
      "  \u001b[0;34minput           : \u001b[0;32msamplesheet.csv\u001b[0m\n",
      "  \u001b[0;34moutdir          : \u001b[0;32mresults_rnaseq\u001b[0m\n",
      "\n",
      "\u001b[1mReference genome options\u001b[0m\n",
      "  \u001b[0;34mgenome          : \u001b[0;32mGRCm38\u001b[0m\n",
      "  \u001b[0;34mfasta           : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Sequence/WholeGenomeFasta/genome.fa\u001b[0m\n",
      "  \u001b[0;34mgtf             : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.gtf\u001b[0m\n",
      "  \u001b[0;34mgene_bed        : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.bed\u001b[0m\n",
      "  \u001b[0;34mstar_index      : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Sequence/STARIndex/\u001b[0m\n",
      "\n",
      "\u001b[1mAlignment options\u001b[0m\n",
      "  \u001b[0;34mmin_mapped_reads: \u001b[0;32m5\u001b[0m\n",
      "\n",
      "\u001b[1mMax job request options\u001b[0m\n",
      "  \u001b[0;34mmax_cpus        : \u001b[0;32m3\u001b[0m\n",
      "  \u001b[0;34mmax_memory      : \u001b[0;32m7.GB\u001b[0m\n",
      "\n",
      "!! Only displaying parameters that differ from the pipeline defaults !!\n",
      "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
      "If you use nf-core/rnaseq for your analysis please cite:\n",
      "\n",
      "* The pipeline\n",
      "  https://doi.org/10.5281/zenodo.1400710\n",
      "\n",
      "* The nf-core framework\n",
      "  https://doi.org/10.1038/s41587-020-0439-x\n",
      "\n",
      "* Software dependencies\n",
      "  https://github.com/nf-core/rnaseq/blob/master/CITATIONS.md\n",
      "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "\u001b[2A\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "\u001b[7A\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "\u001b[10A\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "\u001b[13A\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "\u001b[19A\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "\u001b[28A\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001b[K\n",
      "\u001b[39A\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "\u001b[47A\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001b[K\n",
      "\u001b[48A\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001b[K\n",
      "\u001b[48A\n",
      "executor >  local (1)\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[1f/c2a96e] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001b[K\n",
      "\u001b[49A\n",
      "executor >  local (1)\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[1f/c2a96e] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001b[K\n",
      "\u001b[49A\n",
      "executor >  local (1)\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[1f/c2a96e] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001b[K\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Mus_musculus/Ensembl/GRCm38/Sequence/WholeGenomeFasta/genome.fa\u001b[K\n",
      "\u001b[50A\n",
      "executor >  local (1)\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[1f/c2a96e] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001b[K\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Mus_musculus/Ensembl/GRCm38/Sequence/WholeGenomeFasta/genome.fa\u001b[K\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.gtf\u001b[K\n",
      "\u001b[51A\n",
      "executor >  local (1)\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[1f/c2a96e] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[51A\n"
     ]
    }
   ],
   "source": [
    "# post here the command you used to run nf-core/rnaseq\n",
    "\n",
    "! nextflow run nf-core/rnaseq --input samplesheet.csv --genome GRCm38 --outdir results_rnaseq -profile docker\n",
    "#! nextflow run nf-core/rnaseq -r 3.14.0 --input samplesheet.csv --outdir results_rnaseq --genome GRCm38 -profile docker"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We got the data from the tutors. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<br><br>\n",
    "##### Explain all the parameters you set and why you set them in this way."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- *input*: that contains metadata and file paths for the samples -> pipeline needs to know where the files \n",
    "\n",
    "- *genome*: Specifies the reference genome to use for alignment and analysis -> checking for the genome of the samples \"SRX19144486\",\"SRX19144486\" at NIH gives us GRCm38 \n",
    "\n",
    "- *outdir*: Specifies the output directory where the results of the analysis will be saved\n",
    "\n",
    "- *aligner*: STAR si used for aligning the reads to the reference genome, and Salmon is used for transcript quantification -> efficient combination (fast and little memory)<br><br>\n",
    "\n",
    "**Sources:**\n",
    "\n",
    "Dobin, A., Davis, C.A., Schlesinger, F., Drenkow, J., Zaleski, C., Jha, S., ... & Gingeras, T.R. (2013). STAR: ultrafast universal RNA-seq aligner. Bioinformatics, 29(1), 15-21. https://doi.org/10.1093/bioinformatics/bts635 <br>\n",
    "\n",
    "Patro, R., Duggal, G., Love, M.I., Irizarry, R.A., & Kingsford, C. (2017). Salmon provides fast and bias-aware quantification of transcript expression. Nature Methods, 14(4), 417-419. https://doi.org/10.1038/nmeth.4197\n",
    "<br><br>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Browsing the results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### How did the pipeline perform?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The workflow completed successfully, but the overall runtime of 10 minutes and 1 second is relatively high. The total CPU usage of 151.2 CPU hours highlights the CPU-intensive nature of the tasks involved.<br>\n",
    "\n",
    "Among the processes, STAR-ALIGN-IGENOMES, TRIMGALORE, and SAMTOOLS_SORT stand out with significantly higher CPU and memory usage compared to other processes, suggesting these tasks are particularly resource-intensive. In terms of execution time, both SALMON_INDEX and STAR-ALIGN-IGENOMES show the longest durations.<br>\n",
    "\n",
    "A closer look at the Nextflow Timeline reveals noticeable delays, particularly during the STAR-ALIGN-IGENOMES step.<br>\n",
    "This step appears to be a critical area for performance optimization.<br>\n",
    "\n",
    "In conclusion, the high CPU and memory usage of key processes and the delays shown in the Nextflow Timeline suggest areas for potential optimization, such as improving parallelism and optimizing resource allocation to reduce overall execution time.\n",
    "<br><br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Explain the quality control steps. Are you happy with the quality and why. If not, why not.\n",
    "##### Please give additional information on : ribosomal rRNA, Duplication and GC content\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. FastQC<br>\n",
    "FastQC is a tool used to access the quality of raw sequencing data.<br>\n",
    "Most bases have a high-quality score (Phred > 25). The sequence length distribution (**FastQC: Per Sequence Quality Score**) shows that all samples have sequences of a single length (150bp).<br>\n",
    "Sham_sal_1 and Sham-oxy_1 contain sequences that appear more frequently than expected, which could indicate contamination or sequencing artifacts.<br>\n",
    "While the majority of samples have a very low percentage of **rRNA** (<3%), two samples (SNI_Sal_2 (22%) and SNI_Sal_4 (25%)) exhibit significantly higher rRNA content, which may point to sample-specific issues.<br>\n",
    "The **GC content** for most samples follows a roughly normal distribution (**FastQC: Per Sequence GC Content**). However, some samples, specifically Sham_oxy_1_1 and Sham_oxy_1_2, show a secondary peak toward higher GC percentages. These two samples also have a notably higher GC content compared to the other samples, which could indicate deviations in their composition or potential contamination.<br><br>\n",
    "\n",
    "<div style=\"display: flex; justify-content: space-around;\">\n",
    "    <img src=\"./plots/FastQC/fastqc_per_sequence_quality_scores_plot-2-1.png\" alt=\"Per Sequence Quality Score\" width=\"400\">\n",
    "    <img src=\"./plots/FastQC/fastqc_per_sequence_gc_content_plot-2-1.png\" alt=\"Per Sequence GC Content\" width=\"400\">\n",
    "</div>\n",
    "\n",
    "<br><br>\n",
    "\n",
    "2. RSeQC<br>\n",
    "The **RSeQC Read Distribution** reveals a consistent pattern across all samples, with most reads mapping to exonic regions. These two factors, particularly the high alignment to exonic regions, suggest minimal bias in the data.<br>\n",
    "The **RSeQC Junction Saturation** plot counts the number of known splicing junctions observed in each dataset.<br>\n",
    "The plot indicates that the sequencing depth is adequate, showing a curve that rises steeply at lower depths (<10%) and plateaus at higher depths (>70%), which signifies that most splice junctions have been identified.<br>\n",
    "Furthermore, the plot showing **Splicing Junctions** demonstrates that known splicing junctions are detected in all samples at over 80%, confirming that the sequencing data effectively captures the majority of expected junctions. <br><br>\n",
    "\n",
    "<div style=\"display: flex; justify-content: space-around;\">\n",
    "    <img src=\"./plots/RSeQC/rseqc_read_distribution_plot-1.png\" alt=\"Read Distribution\" width=\"400\">\n",
    "    <img src=\"./plots/RSeQC/rseqc_junction_saturation_plot-1.png\" alt=\"Junction Saturation\" width=\"400\">\n",
    "    <img src=\"./plots/RSeQC/rseqc_junction_annotation_junctions_plot-1.png\" alt=\"Junction Annotation\" width=\"400\">\n",
    "</div>\n",
    "<br><br>\n",
    "\n",
    "3. **Duplication**<br>\n",
    "In this step, the focus is on assessing the number of duplicate reads.<br>\n",
    "The goal is to achieve a low level of duplicates, indicating a diverse library. Most samples exhibit a high percentage of sequences with a low duplication level (<2), which is desirable. <br>\n",
    "However, some samples, particularly Sham_oxy_1, show a significant percentage of sequences with a high duplication level (>1k). <br>\n",
    "The sample distribution in the **DupRead General Model** plot reflects a clear separation between lowly and highly expressed genes.<br>\n",
    "There is a duplication rates for low-expression levels and a more variable, higher duplication rate for highly expressed genes.<br>\n",
    "<br><br>\n",
    "<img src=\"./plots/Duplication/dupradar-plot.png\" alt=\"Duplication\" width=\"400\">\n",
    "<br><br>\n",
    "\n",
    "4. Alignment Quality<br>\n",
    "When aligning the reads to a reference genome we want a high percentage of reads that align uniquely to the reference genome. <br>\n",
    "The results show that all samples have a high percentage of uniquely mapped reads with a percentage >74%. <br>\n",
    "Only the following samples stand out: Sham_oxy_1 (26.1%), Sham_Sal_1 (54.5%)SNI_Sal_2 (55.4%), SNI_Sal_3, SNI_Sal_4.<br><br>\n",
    "In summary, we can say that the data is of good quality as all the metrices show desirable/ expected results.<br>\n",
    "Especially, the per base sequence quality is consistently high and there is no major bias in the gene body coverage indicating good library preparation.<br>\n",
    "The alignment rate is also high with a high number of uniquely mapped reads for most samples.<br>\n",
    "Additionally, the percentage of duplicate reads is relatively low and the distribution of duplicated reads per sample reflect a clear separation between lowly and highly expressed genes.<br>\n",
    "The library seems to have a good diversity and no major issues with with library complexity or technical duplication indicating that  the RNA-Seq dataset is suitable for further analysis<br><br>\n",
    "**Sources:**<br>\n",
    "Wang L, Wang S, Li W* RSeQC: quality control of RNA-seq experiments Bioinformatics (2012) 28 (16): 2184-2185. doi: 10.1093/bioinformatics/bts356 <br>\n",
    "Andrews, S. (2010). FastQC: A Quality Control tool for High Throughput Sequence Data. Available at: http://www.bioinformatics.babraham.ac.uk/projects/fastqc/<br><br>\n",
    "Ewels, P., Magnusson, M., Lundin, S., & Kller, M. (2016). MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics, 32(19), 3047-3048. https://doi.org/10.1093/bioinformatics/btw354\n",
    "<br><br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### What are the possible steps that could lead to poorer results?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Poor RNA-seq results can stem from various steps throughout the process.<br>\n",
    "Issues during sample preparation, including sample collection and library preparation, can significantly impact the quality of the results. <br>\n",
    "Additionally, with four replicates for each condition, variability can be introduced, potentially leading to suboptimal outcomes. <br>\n",
    "Sequencing-related problems, such as low read depth, can also contribute to poor results.<br> Furthermore, during the analysis phase, inadequate quality control or incorrect alignment can further compromise the integrity of the findings.<br><br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Would you exclude any samples? If yes, which and why?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Based on the results of all the quality assessments, it may seem reasonable to consider excluding the samples Sham_oxy_1, SNI_Sal_2, and SNI_Sal_4. However, in a biological experiment, it's important to retain at least one sample from each condition to maintain replicates. Since none of the samples exhibited extraordinarily poor quality control results, we have decided to keep all of them for further analysis.<br><br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### What would you now do to continue the experiment? What are the scientists trying to figure out? Which packages on R or python would you use?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I would proceed with the experiment by conducting secondary analysis, specifically differential expression analysis, to examine the expression levels across different conditions and identify differentially expressed genes (DEGs).<br>\n",
    "This can be accomplished using the Bioconda Python package, the R Bioconductor package DESeq2, or by employing an nf-core pipeline.\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "CompWorkflow",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
